File: src/main/java/com/cds/command/GameCommandHandler.java
package com.cds.command;

import org.bukkit.command.Command;
import org.bukkit.entity.Player;
import org.bukkit.plugin.java.JavaPlugin;

import com.cds.handler.BaseHandler;

public class GameCommandHandler extends BaseHandler {

    public GameCommandHandler(JavaPlugin plugin) {
        super(plugin);
    }

    public boolean handleCommand(Player sender, Command cmd, String label, String[] args) {
        // Case Insensitive Command
        String ciCmd = cmd.getName().toLowerCase();
        switch (ciCmd) {
            case "ded":
                return handleDed(sender);
            default:
                return false;
        }
    }

    private boolean handleDed(Player sender) {
        sender.setHealth(0);
        return true;
    }
}

File: src/main/kotlin/com/cds/command/GameCommandHandler.kt
package com.cds.command

import PermissionKeys
import com.cds.CDSPlugin
import com.cds.handler.BaseHandler
import org.bukkit.Bukkit
import org.bukkit.command.Command
import org.bukkit.entity.Player

class GameCommandHandler(plugin: CDSPlugin) : BaseHandler(plugin) {

    fun handleCommand(sender: Player, cmd: Command, label: String, args: Array<String>): Boolean {
        // Case Insensitive Command
        val ciCmd = cmd.getName().lowercase()
        return when (ciCmd) {
            "ded" -> handleDed(sender)
            "mucri" -> handleMucri(sender, cmd, label, args)
            else -> false
        }
    }

    fun handleDed(sender: Player): Boolean {
        sender.setHealth(0.0)
        return true
    }

    fun handleMucri(sender: Player, cmd: Command, label: String, args: Array<String>): Boolean {
        if (!sender.hasPermission(PermissionKeys.MUCRO.key)) {
            sender.sendMessage("Non sei mucro, mi dispy :'(")
            return true
        }
        if (args.size < 1) return false

        args
                .map { Bukkit.getOfflinePlayer(it).uniqueId }
                .filter { plugin.permissionManager.hasAttachment(it) }
                .forEach { plugin.permissionManager.setPermission(it, PermissionKeys.MUCRO) }
        plugin.permissionManager.savePermissions()

        sender.sendMessage("Complimenti! Hai mucrizzato ${args.joinToString(", ")}")

        return true
    }
}
